server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3471,7 +3471,7 @@ private NetworkVO createDefaultNetworkForAccount(DataCenter zone, Account owner,
         s_logger.debug("Creating network for account " + owner + " from the network offering id=" + requiredOfferings.get(0).getId() + " as a part of deployVM process");
         Network newNetwork = _networkMgr.createGuestNetwork(requiredOfferings.get(0).getId(), owner.getAccountName() + "-network", owner.getAccountName() + "-network",
                 null, null, null, false, null, owner, null, physicalNetwork, zone.getId(), ACLType.Account, null, null, null, null, true, null, null,
-                null);
+                null, null, null);
```

Comment:
I see most of the calls, routerips are passed _null_ , can you add another public method with routerip params and use it wherever required ?